CMakeLists.txt

```diff
@@ -297,6 +297,22 @@ if(MINGW)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mstackrealign -fno-strict-aliasing")
 endif()
 
+# support building on Raspberry Pi 1/2/3 and BBB
+if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "armv7l")
+  foreach(flag CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    set(${flag} "${${flag}} -mtune=cortex-a7")
+    set(${flag} "${${flag}} -mfloat-abi=hard")
+    set(${flag} "${${flag}} -mfpu=neon")
```

Comment:
i had hopes for that `native` flag.  it's a desirable solution but i never got it to work.  tried to compile with it so many times and in different combinations, but somehow it crashed or produced slow binaries.  i have learned a bit more since then so maybe i can help track down the problem now.  i'll give it a go coming days.